vendor/bat-native-ledger/src/bat/ledger/internal/credentials/credentials_sku.cc

```diff
@@ -0,0 +1,373 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <stdint.h>
+
+#include <utility>
+#include <vector>
+
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "bat/ledger/internal/credentials/credentials_sku.h"
+#include "bat/ledger/internal/credentials/credentials_util.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/request/request_sku.h"
+#include "bat/ledger/internal/request/request_util.h"
+#include "bat/ledger/internal/static_values.h"
+#include "net/http/http_status_code.h"
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace {
+
+const char public_key_dev[] =
+    "Tki2ZGeXBlqil4nM1REPA2GAzvVYnnCtyE2ai1HXjEY=";
+const char public_key_staging[] =
+    "rqQ1Tz26C4mv33ld7xpcLhuX1sWaD+s7VMnuX6cokT4=";
+const char public_key_production[] =
+    "rqQ1Tz26C4mv33ld7xpcLhuX1sWaD+s7VMnuX6cokT4=";
+
+bool IsPublicKeyValid(const std::string& public_key) {
+  if (public_key.empty()) {
+    return false;
+  }
+
+  std::string order_key;
+  if (ledger::_environment == ledger::Environment::PRODUCTION) {
+    order_key = public_key_production;
+  }
+
+  if (ledger::_environment == ledger::Environment::STAGING) {
+    order_key = public_key_staging;
+  }
+
+  if (ledger::_environment == ledger::Environment::DEVELOPMENT) {
+    order_key = public_key_dev;
+  }
+
+  return order_key == public_key;
+}
+
+std::string ConvertItemTypeToString(const std::string& type) {
+  int type_int;
+  base::StringToInt(type, &type_int);
+  switch (static_cast<ledger::SKUOrderItemType>(type_int)) {
+    case ledger::SKUOrderItemType::SINGLE_USE: {
+      return "single-use";
+    }
+    case ledger::SKUOrderItemType::NONE: {
+      return "";
+    }
```

Comment:
Return "" for default case

Reply:
we try to ignore default case as then if we add new property to enum it will not pick up in switch that we need to handle